Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPathEditor.cs

```diff
@@ -200,7 +195,16 @@ private void SetExpectedControlLayoutFromAttribute(SerializedProperty property)
                 SetExpectedControlLayout(attribute.layout);
         }
 
-        public SerializedProperty pathProperty { get; }
+        // This static variable is a hack. Because the editor is rebuilt at unpredictable times with a new serializedObject, we need to keep updating
+        // this variable with most up to date serializedProperty, so that the picker dropdown can access the correct serializedProperty.
+        // The picker dropdown is a separate window and does not have access to the changed serializedObject reference.
+        // This could be removed if the InputControlPathEditor is converted to UITK with a stable, persistent serializedObject backing this editor.
+        // This property will be shared among multiple asset editor windows.
+        private static SerializedProperty s_pathProperty { get; set; }
```

Comment:
This static SerializedProperty, wouldn't it leak memory? Its referencing a SerializedObject that was dynamically allocated and may/would be disposed from outside this editor?

Reply:
Additionally this assumes there can only exist one InputControlPathEditor instance right?

Reply:
Yes, there will be a dangling reference tht may hang around. I'll add some code to clean it up when the editor is closed.

Yes, only one instance of the InputControlPathEditor will work at a time. I think this is ok, as it is a popup window, and disappears when it loses focus.

Reply:
https://jira.unity3d.com/browse/ISX-2257